policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +68,52 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for container in spec.iter().flat_map(|s| s.containers.iter()) {
+        set_probe_ports(&container.liveness_probe, &mut probes, port_names);
+        set_probe_ports(&container.readiness_probe, &mut probes, port_names);
+        set_probe_ports(&container.startup_probe, &mut probes, port_names);
+    }
+    probes
+}
+
+fn set_probe_ports(
+    probe: &Option<k8s::Probe>,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    let probe = match probe {
+        Some(probe) => probe,
+        None => return,
+    };
+    if let Some(http) = &probe.http_get {
+        if let Some(path) = &http.path {
+            match &http.port {
+                k8s::IntOrString::Int(port) => {
+                    if let Ok(port) = u16::try_from(*port).and_then(NonZeroU16::try_from) {
+                        let paths = probes.entry(port).or_default();
+                        paths.insert(path.clone());
+                    }
+                }
+                k8s::IntOrString::String(name) => {
+                    for port in port_names.get(name).into_iter().flatten() {
+                        let paths = probes.entry(*port).or_default();
+                        paths.insert(path.clone());
+                    }
+                }
```

Comment:
Also, I know this is part of the k8s spec but I'm curious what actually happens if you define a probe that references multiple ports via name...

Reply:
So, now that I think about this -- I think this is wrong (as is my suggestion below, somewhat):

This name must refer to a port _in the same container_, not on any container in the pod. And names must be unique within a single container.